lambdas/retrieve-cri-credential/src/main/java/uk/gov/di/ipv/core/retrievecricredential/RetrieveCriCredentialHandler.java

```diff
@@ -168,50 +167,51 @@ public Map<String, Object> handleRequest(Map<String, String> input, Context cont
                             ? configService.getCriPrivateApiKey(credentialIssuerId)
                             : null;
 
+            final AuditEventUser auditEventUser =
+                    new AuditEventUser(
+                            userId,
+                            ipvSessionItem.getIpvSessionId(),
+                            clientOAuthSessionItem.getGovukSigninJourneyId(),
+                            ipAddress);
+
             VerifiableCredentialResponse verifiableCredentialResponse =
                     verifiableCredentialService.getVerifiableCredentialResponse(
                             BearerAccessToken.parse(criOAuthSessionItem.getAccessToken()),
                             credentialIssuerConfig,
                             apiKey,
                             credentialIssuerId);
 
-            auditEventUser =
-                    new AuditEventUser(
-                            userId,
-                            ipvSessionItem.getIpvSessionId(),
-                            clientOAuthSessionItem.getGovukSigninJourneyId(),
-                            ipAddress);
-
             if (VerifiableCredentialStatus.PENDING.equals(
                     verifiableCredentialResponse.getCredentialStatus())) {
+                sendCriResRetrievedAuditEvent(
+                        auditEventUser,
+                        credentialIssuerId,
+                        CriResourceRetrievedType.PENDING.getType());
                 processPendingResponse(
                         userId,
                         credentialIssuerId,
                         credentialIssuerConfig,
                         verifiableCredentialResponse,
                         criOAuthSessionItem.getCriOAuthSessionId(),
                         ipvSessionItem);
-                sendCriResRetrievedAuditEvent(
-                        auditEventUser,
-                        credentialIssuerId,
-                        CriResourceRetrievedType.PENDING.getType());
             } else {
                 List<SignedJWT> verifiableCredentials =
                         verifiableCredentialResponse.getVerifiableCredentials();
+                if (!verifiableCredentials.isEmpty()) {
```

Comment:
I'm still unsure about this - if the VC bundle is empty, I think we've still fetched a CRI resource, it's just empty.

It would be good to find out where this requirement came from, to determine what to do - I've asked in the thread to see if Simon knows.

Reply:
Yes agree we got the response with empty vc's. Here we can send type as 'empty' / 'vc-empty'.

As we are sending this event - I think we should decide what and when to pass. I'd checked with TxMA side they also told its team to decide what they want to pass.

Reply:
Ah great - in that case I think we _should_ send an event when it's empty, but with an empty type as you suggest :)

Reply:
done